### PR TITLE
Add Bastion65 PCB to QMK

### PR DESCRIPTION
**Pull request by awkannan**
## Description

Add Bastion65 PCB to QMK

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by dunk2k** [comment]
Quick check as an image isn't provided: is the 1 layout defined here the only physical key arrangement possible on this PCB?
If not, would you be able to add layouts to `info.json` for the other layouts/key combinations?

File: keyboards/cannonkeys/bastion65/info.json [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        "breathing_period": 5,
+        "levels": 15,
+        "pin": "GP22"
+    },
+    "ws2812": {"pin": "GP7"},
+    "rgblight": {
+        "led_count": 20,
+        "hue_steps": 17,
+        "saturation_steps": 17,
+        "brightness_steps": 17,
```
Comment (waffle87) [inline keyboards/cannonkeys/bastion65/info.json:43]:
```suggestion
```

File: keyboards/cannonkeys/bastion65/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT(
+    QK_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,                 KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,   KC_BSPC, KC_DEL, KC_HOME,
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,   KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,  KC_BSLS, KC_END,
+    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,                 KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,           KC_ENT, KC_PGUP,
+    KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,                 KC_M,    KC_COMM, KC_DOT,  KC_SLSH,           KC_RSFT, KC_UP, KC_PGDN,
+    KC_LCTL, KC_LGUI, KC_LALT,                   KC_SPC,  KC_SPC,  KC_SPC,                                          KC_RALT, KC_RGUI, KC_LEFT, KC_DOWN, KC_RIGHT
+  ),
+
+  [_FN1] = LAYOUT(
```
Comment (waffle87) [inline keyboards/cannonkeys/bastion65/keymaps/default/keymap.c:38]: This layer doesn't seem to be accessible.

File: keyboards/cannonkeys/bastion65/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [_BASE] = LAYOUT(
+    QK_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,                 KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,   KC_BSPC, KC_DEL, KC_HOME,
+    KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,   KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,  KC_BSLS, KC_END,
+    KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,                 KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,           KC_ENT, KC_PGUP,
+    KC_LSFT, KC_BSLS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,                 KC_M,    KC_COMM, KC_DOT,  KC_SLSH,           KC_RSFT, KC_UP, KC_PGDN,
+    KC_LCTL, KC_LGUI, KC_LALT,                   KC_SPC,  KC_SPC,  KC_SPC,                                          KC_RALT, KC_RGUI, KC_LEFT, KC_DOWN, KC_RIGHT
+  ),
+
+  [_FN1] = LAYOUT(
```
Comment (waffle87) [inline keyboards/cannonkeys/bastion65/keymaps/via/keymap.c:38]: As above.

File: keyboards/cannonkeys/bastion65/rules.mk [hunk 4]
```diff
@@ -0,0 +1,3 @@
+EEPROM_DRIVER = wear_leveling
+WEAR_LEVELING_DRIVER = rp2040_flash
+WS2812_DRIVER = vendor
```
Comment (waffle87) [inline keyboards/cannonkeys/bastion65/rules.mk:3]:
```suggestion
# This file intentionally left blank
```

File: keyboards/cannonkeys/bastion65/info.json [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        "caps_lock": "GP29",
+        "on_state": 0
+    },
+    "backlight": {
+        "breathing": true,
+        "breathing_period": 5,
+        "levels": 15,
+        "pin": "GP22"
+    },
+    "ws2812": {"pin": "GP7"},
```
Comment (waffle87) [inline keyboards/cannonkeys/bastion65/info.json:38]:
```suggestion
    "ws2812": {
        "pin": "GP7",
        "driver": "vendor"
    },
```

File: keyboards/cannonkeys/bastion65/keymaps/default/keymap.c (hunk 2 above)
Comment (awkannan) [reply keyboards/cannonkeys/bastion65/keymaps/default/keymap.c:38]: Thanks for this call-out - I'll fix!

**Comment by awkannan** [comment]
@waffle87 I've addressed the feedback and this is ready for further review!
(Also, please let me know if you'd prefer if I didn't ping like this)

**Review by waffle87 — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**